.github/workflows/test.yaml

```diff
@@ -3,11 +3,14 @@ on:
   pull_request:
     branches: [ "master", "release/**" ]
   push:
-    branches: [ "release/**" ]
+    # we trigger runs on master branch, but we do not run spread on master 
+    # branch, the master branch runs are just for unit tests + codecov.io
+    branches: [ "master","release/**" ]
 
 jobs:
   snap-builds:
     runs-on: ubuntu-20.04
+    if: ${{ github.event_name != 'push' && github.ref != 'refs/heads/master' }}
```

Comment:
Having a snapd snap built and pushed to edge outweighs keeping some number of snap artifacts so it's good this is disabled when pushing to master.